homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -446,6 +522,48 @@ def native_value(self) -> int | None:
         return self.entity_description.value_fn(system_consumption)
 
 
+class EnvoyProductionPhaseEntity(EnvoySystemSensorEntity):
+    """Envoy phase production entity."""
+
+    entity_description: EnvoyProductionSensorEntityDescription
+
+    @property
+    def native_value(self) -> int | None:
+        """Return the state of the sensor."""
+        if (
+            self.entity_description.on_phase is None
+            or self.data.system_production_phases is None
+        ):
+            return None
+        system_production = self.data.system_production_phases[
+            self.entity_description.on_phase
+        ]
+        if system_production is None:
+            return None
+        return self.entity_description.value_fn(system_production)
+
+
+class EnvoyConsumptionPhaseEntity(EnvoySystemSensorEntity):
+    """Envoy phase consumption entity."""
+
+    entity_description: EnvoyConsumptionSensorEntityDescription
+
+    @property
+    def native_value(self) -> int | None:
+        """Return the state of the sensor."""
+        if (
+            self.entity_description.on_phase is None
+            or self.data.system_consumption_phases is None
+        ):
+            return None
+        system_consumption = self.data.system_consumption_phases[
+            self.entity_description.on_phase
+        ]
+        if system_consumption is None:
```

Comment:
Its already pretty long but I still think you save a line by using a walrus here